lib/all_changed_attributes.rb

```diff
@@ -0,0 +1,47 @@
+require 'all_changed_attributes/version'
+
+module AllChangedAttributes
+  def write_attribute(attr_name, value)
+    store_all_changed_attributes(attr_name, value) if attributes[attr_name] != value
+    super
+  end
+
+  def all_changed_attributes(new = {})
+    new.merge(@all_changed_attributes ||= {})
+  end
+
+  def store_all_changed_attributes(attr_name, value)
+    @all_changed_attributes = all_changed_attributes({attr_name.to_s => attributes[attr_name]})
+  end
+
+  if ActiveRecord::VERSION::MAJOR > 4 || (ActiveRecord::VERSION::MAJOR == 4 && ActiveRecord::VERSION::MINOR > 0)
```

Comment:
I've dropped support for these old versions of rails from ar_transaction_changes and identity_cache